app/models/stage_collection.rb

```diff
@@ -11,9 +11,9 @@ def initialize(stages, object)
     @stages = initialize_stages(stages)
   end
 
-  def each(&_block)
+  def each(&block)
```

Comment:
I wonder if we could make this even simpler by removing this block entirely and instead delegating to @stages at line 7

```
  def_delegators :@stages, :each, :second, :last, :size
```

and spec

```
  describe '#each' do
    it 'yields successive stages' do
      expect { |b| collection.each(&b) }.to yield_successive_args(kind_of(Stage), kind_of(Stage))
    end
  end
```